primitives/sidechain/src/traits/mod.rs

```diff
@@ -41,7 +40,7 @@ pub trait Header: Encode + Decode + Clone {
 
 	/// get the `blake2_256` hash of the header.
 	fn hash(&self) -> H256 {
-		self.using_encoded(blake2_256).into()
+		self.using_encoded(BlakeTwo256::hash).into()
```

Comment:
Same function, but only `BlakeTwo256` is available in no_std, without activating the `full_crypto` feature.